internal/sql/repository/chartConfig/PipelineConfigRepository.go

```diff
@@ -59,33 +59,41 @@ func (impl PipelineConfigRepositoryImpl) Save(pipelineStrategy *PipelineStrategy
 }
 
 func (impl PipelineConfigRepositoryImpl) Update(pipelineStrategy *PipelineStrategy, tx *pg.Tx) error {
-	_, err := impl.dbConnection.Model(pipelineStrategy).WherePK().UpdateNotNull()
+	_, err := tx.Model(pipelineStrategy).WherePK().UpdateNotNull()
 	return err
 }
 
 func (impl PipelineConfigRepositoryImpl) FindById(id int) (pipelineStrategy *PipelineStrategy, err error) {
 	pipelineStrategy = &PipelineStrategy{}
 	err = impl.dbConnection.Model(pipelineStrategy).
-		Where("id = ?", id).Select()
+		Where("id = ?", id).
+		Where("deleted = ?", false).
+		Select()
 	return pipelineStrategy, err
 }
 
 func (impl PipelineConfigRepositoryImpl) FindByStrategy(strategy chartRepoRepository.DeploymentStrategy) (pipelineStrategy *PipelineStrategy, err error) {
 	pipelineStrategy = &PipelineStrategy{}
 	err = impl.dbConnection.Model(pipelineStrategy).
-		Where("strategy = ?", strategy).Select()
+		Where("strategy = ?", strategy).
+		Where("deleted = ?", false).
+		Select()
 	return pipelineStrategy, err
 }
 
 func (impl PipelineConfigRepositoryImpl) FindByStrategyAndPipelineId(strategy chartRepoRepository.DeploymentStrategy, pipelineId int) (pipelineStrategy *PipelineStrategy, err error) {
```

Comment:
have we checked all current usages in both oss and enterprise and sure this wont break